clang/test/CIR/Lowering/ThroughMLIR/if.cir

```diff
@@ -0,0 +1,63 @@
+// RUN: cir-opt %s -cir-to-mlir -o %t.mlir
+// RUN: FileCheck %s --input-file %t.mlir
```

Comment:
Same comment from other PR: please add a comment with the C/C++ source to regen this or/and add a testcase coming from C/C++. I'd prefer if going fwd we start using more source code instead: (a) easier to update and (b) target triples and other informations that will be useful at some point.

Reply:
Ok, I will add some test cases and comments

Reply:
Hello, I add a  C source test, can you review it?

Reply:
Great thanks. Please delete this one then!